static/app/components/core/slider/index.chonk.tsx

```diff
@@ -0,0 +1,317 @@
+import {
+  type CSSProperties,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import type {DO_NOT_USE_ChonkTheme} from '@emotion/react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import {mergeRefs} from '@react-aria/utils';
+
+import {space} from 'sentry/styles/space';
+
+import type {SliderProps} from './index';
+
+const passthrough = (n: number | '') => n;
+
+export function Slider({
+  formatLabel = passthrough,
+  ref,
+  ...props
+}: SliderProps & {ref: React.Ref<HTMLInputElement>}) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const refs = mergeRefs(ref, inputRef);
+  const step = toNumber(props.step ?? -1);
+  const {value, min, max} = resolveMinMaxValue(props);
+  const [label, setLabel] = useState(value);
+  const initialProgress = getProgress(value, min, max);
+  const filledSteps = useMemo(() => (step === -1 ? -1 : label / step), [step, label]);
+
+  const handleChange = useCallback(
+    (event: Event) => {
+      const input = event.target as HTMLInputElement;
+      const {valueAsNumber, min: nativeMin, max: nativeMax} = input;
+      setLabel(valueAsNumber);
+      const progress = getProgress(valueAsNumber, nativeMin || 0, nativeMax || 100);
+      input.parentElement?.style.setProperty('--p', `${progress.toFixed(0)}%`);
+    },
+    [setLabel]
+  );
+
+  useEffect(() => {
+    const input: HTMLInputElement | null = inputRef.current;
+    if (!input) {
+      return;
+    }
+    input.addEventListener('input', handleChange);
+    // eslint-disable-next-line consistent-return
+    return () => input.removeEventListener('input', handleChange);
+  });
+
+  return (
+    <SliderContainer
+      aria-disabled={props.disabled}
+      style={
+        {'--p': `${initialProgress.toFixed(0)}%`, '--steps': `${label}`} as CSSProperties
+      }
+    >
+      {props.step ? (
+        <SliderTicks
+          filledSteps={filledSteps}
+          n={
+            typeof props.step === 'string' ? Number.parseInt(props.step, 10) : props.step
+          }
+        />
+      ) : null}
+      <StyledSlider ref={refs} type="range" {...props} />
+      {props.disabled ? null : (
+        <SliderOutput htmlFor={props.id}>
+          <SliderLabel>{formatLabel(label)}</SliderLabel>
+        </SliderOutput>
+      )}
+    </SliderContainer>
+  );
+}
+
+function SliderTicks({
+  n,
+  filledSteps,
+}: {
+  filledSteps: number;
+  n: number;
+  disabled?: boolean;
+}) {
+  return (
+    <StepsContainer role="presentation">
+      {Array.from({length: n}, (_, i) => {
+        return <StepMark key={i} filled={i <= filledSteps} />;
+      })}
+    </StepsContainer>
+  );
+}
+
+const StepsContainer = styled('div')<{theme?: DO_NOT_USE_ChonkTheme}>`
```

Comment:
This will give you the correct theme out of the box
```suggestion
const StepsContainer = chonkStyles('div')`
```